game/src/Game/MusicConductor/MusicConductor.gd

```diff
@@ -67,24 +73,78 @@ func start_song_by_index(id: int) -> void:
 # REQUIREMENTS
 # * SS-69
 func select_next_song() -> void:
-	_selected_track = (_selected_track + 1) % len(_available_songs)
+	#_selected_track = (_selected_track + 1) % len(_available_songs)
+	if playlist_index >= playlist_len or playlist_index >= len(playlist):
+		generate_playlist()
+		playlist_index = 0
+	_selected_track = playlist[playlist_index]
+	playlist_index += 1
+	last_played = playlist_index
 	start_current_song()
 
 func select_previous_song() -> void:
 	_selected_track = (len(_available_songs) - 1) if (_selected_track == 0) else (_selected_track - 1)
 	start_current_song()
 
-# REQUIREMENTS
-# * SND-2, SND-3
-func _ready() -> void:
+func setup_compat_song(file_name) -> void:
+	var song = SongInfo.new()
+	var stream = SoundSingleton.get_song(file_name)
+
+	var metadata = MusicMetadata.new()
+	metadata.set_from_stream(stream)
+	var title = metadata.title
+	if title == "":
+		#use the file name without the extension if there's no metadata
+		title = file_name.split(".")[0]
+	song.init_stream(file_name,title,stream)
+	_available_songs.append(song)
+	
+func add_compat_songs() -> void:
+	for file_name : String in SoundSingleton.song_list:
+		setup_compat_song(file_name)
+
+func add_ootb_music() -> void:
 	var dir := DirAccess.open(music_directory)
 	for fname : String in dir.get_files():
 		if fname.ends_with(".import"):
 			fname = fname.get_basename()
 			if fname.get_basename() == first_song_name:
 				_selected_track = _available_songs.size()
-			_available_songs.append(SongInfo.new(music_directory, fname))
-	start_current_song()
+			var song = SongInfo.new()
+			song.init_file_path(music_directory, fname)
+			_available_songs.append(song)
+
+func generate_playlist() -> void:
+	var song_names = MusicConductor.get_all_song_names()
+	var possible_indices = range(len(song_names)-1)
+	
+	var title_index = song_names.find("The Coronation - Title")
+	possible_indices.remove_at(title_index)
+	
+	if len(possible_indices) < playlist_len:
+		playlist_len = possible_indices
+	
+	if len(playlist) != playlist_len:
+		playlist.resize(playlist_len)
+		playlist.fill(0)
+
+	for i in range(playlist_len):
+		var ind = randi_range(0,len(possible_indices)-1)
+		#The song we just played can be in the playlist, just not the first one
+		if i==0:
+			while(ind == last_played):
+				ind = randi_range(0,len(possible_indices)-1)
```

Comment:
This technically could cause an infinite loop, although very unlikely in practice of course.